tests/test_requests.py

```diff
@@ -2585,5 +2585,18 @@ def test_post_json_nan(self, httpbin):
 
     def test_json_decode_compatibility(self, httpbin):
         r = requests.get(httpbin('bytes/20'))
-        with pytest.raises(requests.exceptions.JSONDecodeError):
+        data = r.text
+        with pytest.raises(requests.exceptions.JSONDecodeError) as excinfo:
             r.json()
+        assert isinstance(excinfo.value, RequestException)
+        assert isinstance(excinfo.value, JSONDecodeError)
+        assert data not in str(excinfo.value)
+
+    @pytest.mark.skipif(not is_py3, reason="doc attribute is only present on py3")
+    def test_json_decode_persists_doc_attr(self, httpbin):
+        r = requests.get(httpbin('bytes/20'))
+        data = r.text
```

Comment:
nit; since we're only using this once we can probably just use r.text in the final assertion as shown in the original proposal.